apps/dashboard/src/components/workflow-editor/steps/use-ai-suggestions.tsx

```diff
@@ -0,0 +1,235 @@
+import { useFeatureFlag } from '@/hooks/use-feature-flag';
+import { FeatureFlagsKeysEnum, StepResponseDto, StepTypeEnum, WorkflowResponseDto } from '@novu/shared';
+import { useCallback, useMemo, useState } from 'react';
+import { useFormContext } from 'react-hook-form';
+import { v4 as uuidv4 } from 'uuid';
+
+const AI_SUPPORTED_STEP_TYPES = [
+  StepTypeEnum.EMAIL,
+  StepTypeEnum.SMS,
+  StepTypeEnum.PUSH,
+  StepTypeEnum.IN_APP,
+  StepTypeEnum.CHAT,
+];
+
+function inferTypeFromValue(value: any): 'string' | 'number' | 'boolean' | 'object' | 'array' {
+  if (typeof value === 'string') return 'string';
+  if (typeof value === 'number') return 'number';
+  if (typeof value === 'boolean') return 'boolean';
+  if (Array.isArray(value)) return 'array';
+  if (typeof value === 'object' && value !== null) return 'object';
+  return 'string';
+}
+
+function getSampleValueForVariable(varName: string): string {
+  const lowerName = varName.toLowerCase();
+  if (lowerName.includes('url') || lowerName.includes('link')) {
+    return 'https://example.com';
+  }
+  if (lowerName.includes('email')) {
+    return 'user@example.com';
+  }
+  if (lowerName.includes('name') || lowerName.includes('company')) {
+    return 'Acme Inc';
+  }
+  if (lowerName.includes('order') || lowerName.includes('number') || lowerName.includes('id')) {
+    return '#12345';
+  }
+  if (lowerName.includes('amount') || lowerName.includes('price') || lowerName.includes('total')) {
+    return '$99.00';
+  }
+  if (lowerName.includes('date') || lowerName.includes('time')) {
+    return new Date().toLocaleDateString();
+  }
+  return `sample_${varName}`;
+}
+
+interface UseAiSuggestionsProps {
+  workflow: WorkflowResponseDto;
+  step: StepResponseDto;
+  editorValue: string;
+  setEditorValue: (value: string) => void;
+}
+
+export function useAiSuggestions({ workflow, step, editorValue, setEditorValue }: UseAiSuggestionsProps) {
+  const form = useFormContext();
+  const [isAiDialogOpen, setIsAiDialogOpen] = useState(false);
+  const { addProperty, getSchemaPropertyByKey, isPayloadSchemaEnabled } = useWorkflowSchema();
+  const isAiStepGenerationEnabled = useFeatureFlag(FeatureFlagsKeysEnum.IS_AI_STEP_GENERATION_ENABLED);
+
+  const showAiButton = isAiStepGenerationEnabled && AI_SUPPORTED_STEP_TYPES.includes(step.type as StepTypeEnum);
+
+  const hasExistingContent = useMemo(() => {
+    const formValues = form.getValues();
+    switch (step.type) {
+      case StepTypeEnum.EMAIL:
+        return !!(formValues.subject || formValues.body);
+      case StepTypeEnum.SMS:
+      case StepTypeEnum.CHAT:
+        return !!formValues.body;
+      case StepTypeEnum.PUSH:
+        return !!(formValues.subject || formValues.body);
+      case StepTypeEnum.IN_APP:
+        return !!(formValues.subject || formValues.body);
+      default:
+        return false;
+    }
+  }, [form, step.type]);
+
+  const handleAiInsert = useCallback(
```

Comment:
it should also work with the HTML editor:

https://github.com/user-attachments/assets/34a49714-628b-4036-9cfa-187662c536c7